// Proto file representing the BoardBendDef class

syntax = "proto3";

package ansys.api.edb.v1;

import "edb_messages.proto";
import "point_data.proto";
import "polygon_data.proto";

service BoardBendDefService {
  rpc Create (BoardBendDefCreateMessage) returns (EDBObjMessage) {}

  rpc GetBoundaryPrim (EDBObjMessage) returns (EDBObjMessage) {}

  rpc GetBendMiddle (EDBObjMessage) returns (PointPairMessage) {}

  rpc SetBendMiddle (PointPairPropertyMessage) returns (google.protobuf.Empty) {}

  rpc GetRadius (EDBObjMessage) returns (ValueMessage) {}

  rpc SetRadius (ValuePropertyMessage) returns (google.protobuf.Empty) {}

  rpc GetAngle (EDBObjMessage) returns (ValueMessage) {}

  rpc SetAngle (ValuePropertyMessage) returns (google.protobuf.Empty) {}

  rpc GetBentRegions (EDBObjMessage) returns (PolygonDataListMessage) {}
}

message BoardBendDefCreateMessage {
  EDBObjMessage layout = 1;
  EDBObjMessage zone_prim = 2;
  PointPairMessage middle = 3;
  ValueMessage radius = 4;
  ValueMessage angle = 5;
}
